crud/src/main/java/com/redhat/lightblue/rest/crud/health/ReadHealthCheck.java

```diff
@@ -24,10 +24,16 @@ protected Result check() throws Exception {
         RequestMetrics metrics = new DropwizardRequestMetrics(MetricRegistryFactory.getJmxMetricRegistry());
         CallStatus callStatus = new FindCommand(entity, version, request, metrics).run();
 
-        if(!Response.Status.OK.equals(callStatus.getHttpStatus())){
-            return Result.unhealthy("Reads are not healthy: " + callStatus.getReturnValue());
+        ResultBuilder resultBuilder;
+
+        if (Response.Status.OK.equals(callStatus.getHttpStatus())) {
+            resultBuilder = Result.builder().healthy();
+        } else {
+            resultBuilder = Result.builder().unhealthy().withMessage("Reads are not healthy");
         }
-        return Result.healthy();
+        resultBuilder.withDetail("readResponse", callStatus.getReturnValue());
```

Comment:
Does this serialize as expected? I expected you would have to use `callStatus.toJson()` which is what the regular responses uses (via toString() but point is RESTEasy isn't typically doing the marshalling for us so I'm not sure the call status bodies will all be serializable out of the box, I could be wrong tho). Possible to test?

Reply:
So I did actually mean to use the toJson() here, so that was an oversight, but this also required a change in CrudCheckResponse.  The output now looks like this for healthy..

```
{
    "readHealthCheck": {
        "healthy": true,
        "message": null,
        "readResponse": {
            "entity": "test",
            "entityVersion": "1.0.0",
            "status": "COMPLETE",
            "modifiedCount": 0,
            "matchCount": 0,
            "hostname": "localhost",
            "processed": [],
            "resultMetadata": []
        }
    }
}
```
and like this for unhealthy
```
{
    "readHealthCheck": {
        "healthy": false,
        "message": "Reads are not healthy",
        "readResponse": {
            "entity": "test",
            "entityVersion": "1.0.0",
            "status": "ERROR",
            "modifiedCount": 0,
            "matchCount": 0,
            "hostname": "localhost",
            "errors": [
                {
                    "objectType": "error",
                    "context": "rest/FindCommand/test/find(test:1.0.0)/mongo:find",
                    "errorCode": "mongo-crud:ConnectionError",
                    "msg": "Prematurely reached end of stream"
                }
            ]
        }
    }
}
```